docs/source/setup_marathon_job.rst

```diff
@@ -0,0 +1,31 @@
+setup_marathon_job
+======================================
+
+``setup_marathon_job`` iterates over a list of service.instances to take
+advantage of the installed requests_cache to avoid fetching the mesos
+master state on each service.instance. Each service.instance is called
+an app in ``setup_marathon_job``. Since apps are independent to each other,
+it won't be a problem to cache the mesos master state which may become
+stale when apps are created or destroyed while ``setup_marathon_job`` is
+making progress.
+
+``setup_marathon_job`` grabs a zookeeper lock for each app while it is
+being deployed. Locking is not necessary if `deploy_marathon_services <deploy_marathon_services.html>`_
+is the only source of service deployment. It provides safety against another
+``setup_marathon_job`` being launched, e.g. manually by a system admin.
+
+It takes multiple ``setup_marathon_job`` runs to advance an app from its
+current state to the desired state. For example, below steps occur in
+sequential ``setup_marathon_job`` runs in a crossover bounce:
+
+* The new app is created with marathon and marathon is working to launch
```

Comment:
I wouldn't duplicate these docs here, instead link to the existing bounce documentation